manta-js/package/sdk.ts

```diff
@@ -0,0 +1,604 @@
+import { ApiPromise, WsProvider } from '@polkadot/api';
+import { base58Decode, base58Encode } from '@polkadot/util-crypto';
+// @ts-ignore
+import Api, {ApiConfig} from './api/index';
+import axios from 'axios';
+import BN from 'bn.js';
+import config from './manta-config.json';
+import { Transaction, Wallet } from 'manta-wasm-wallet';
+import { Signer, SubmittableExtrinsic } from '@polkadot/api/types';
+import { Version, Address, AssetId, InitApiResult, InitWasmResult, IMantaPrivateWallet, SignedTransaction } from "./sdk.interfaces";
+
+const rpc = config.RPC;
+const types = config.TYPES;
+const DEFAULT_PULL_SIZE = config.DEFAULT_PULL_SIZE;
+const SIGNER_URL = config.SIGNER_URL;
+const PRIVATE_ASSET_PREFIX = "zk";
+const NATIVE_TOKEN_ASSET_ID = "1";
+
+/// The Envrionment that the sdk is configured to run for, if development
+/// is selected then it will attempt to connect to a local node instance.
+/// If production is selected it will attempt to connect to actual node.
+export enum Environment {
+    Development = "DEV",
+    Production = "PROD"
+}
+
+/// Supported networks.
+export enum Network {
+    Dolphin = "Dolphin",
+    Calamari = "Calamari",
+    Manta = "Manta"
+}
+
+/// MantaPrivateWallet class
+export class MantaPrivateWallet implements IMantaPrivateWallet {
+
+    api: ApiPromise;
+    wasm: any;
+    wasmWallet: Wallet;
+    network: Network;
+    environment: Environment;
+    wasmApi: any;
+    walletIsBusy: boolean;
+    initialSyncIsFinished: boolean;
+    loggingEnabled: boolean;
+
+    constructor(api: ApiPromise, wasm: any, wasmWallet: Wallet, network: Network, environment: Environment, wasmApi: any, loggingEnabled:boolean) {
+        this.api = api;
+        this.wasm = wasm;
+        this.wasmWallet = wasmWallet;
+        this.network = network;
+        this.environment = environment;
+        this.wasmApi = wasmApi;
+        this.walletIsBusy = false;
+        this.initialSyncIsFinished = false;
+        this.loggingEnabled = loggingEnabled;
+    }
+
+    ///
+    /// Public Methods
+    ///
+
+    /// Initializes the MantaPrivateWallet class, for a corresponding environment and network.
+    static async init(env: Environment, network: Network, loggingEnabled:boolean=false, maxReceiversPullSize:number=DEFAULT_PULL_SIZE, maxSendersPullSize:number=DEFAULT_PULL_SIZE, pullCallback:any=null, errorCallback:any=null): Promise<MantaPrivateWallet> {
```

Comment:
Instead of passing all of these parameters separately, it's more convenient to pass in an ApiConfig.